### PR TITLE
utility functions for commonly used steps

### PR DESCRIPTION
**Pull request by minrk**
My first jsonnet functions! 🎉 

- onComponentLabel for adding the frequent `* on (namespace, pod) kube_pod_labels{...)` join to any metric
- cpuResourcePanel, memoryResourcePanel for easily making cpu/memory panels for different components. A `multi` boolean flag creates a panel for all components (with optional exclusions)
- use standardDims properties for width/height in layout, make multi-component resource panels 50% taller to accommodate legend table
- put distribution heatmaps next to each other in usage stats by moving users-per-node next to total users

Screenshot of new resource panels created by these 6 lines:

```jsonnet
local proxyMemory = memoryPanel('Proxy', component='proxy');
local proxyCPU = cpuPanel('Proxy', component='proxy');
local hubMemory = memoryPanel('Hub', component='hub');
local hubCPU = cpuPanel('Hub', component='hub');

// with multi=true, component='singleuser-server' means all components *except* singleuser-server
local allComponentsMemory = memoryPanel('All JupyterHub Components', component='singleuser-server', multi=true);
local allComponentsCPU = cpuPanel('All JupyterHub Components', component='singleuser-server', multi=true);

```

Screen Shot 2021-06-23 at 15 06 36

### CODE REVIEW

File: dashboards/jupyterhub.jsonnet [hunk 1]
```diff
@@ -342,45 +381,45 @@ dashboard.new(
 ).addPanel(
   userNodes, { x: 0, y: 0 } + standardDims
 ).addPanel(
-  clusterUtilization, { x: 12, y: 0 } + standardDims
+  clusterUtilization, { x: standardDims.w, y: 0 } + standardDims
```
Comment (yuvipanda) [inline dashboards/jupyterhub.jsonnet:384]: We can pass just `{}` (or just a `y` or `x`) to these - `deploy.py` autolayouts now.

File: dashboards/jupyterhub.jsonnet [hunk 2]
```diff
@@ -30,9 +30,58 @@ local templates = [
...
+ * Prometheus query for selecting pods
+ *
+ * @name componentLabel
+ *
+ * @param component The component being selected
+ * @param cmp (default `'='`) The comparison to use for the component label, e.g. for exclusions.
+ *
+ * @return prometheus query string selecting pods for a certain component
+ */
+local componentLabel(component, cmp='=') =
```
Comment (yuvipanda) [inline dashboards/jupyterhub.jsonnet:43]: Let's put functions in a separate jsonnet file and import them? Maybe we can reuse them in `cluster.jsonnet` too that way?

**Review by yuvipanda — CHANGES_REQUESTED** [review]
\o/ FUNCTIONS!

File: dashboards/jupyterhub.jsonnet (hunk 2 above)
Comment (minrk) [reply dashboards/jupyterhub.jsonnet:43]: Now I get to learn how imports work!

**Comment by minrk** [comment]
Moved functions to jupyterhub.libsonnet, removed explicit dimensions in layout everywhere defaults are used

**Comment by yuvipanda** [comment]
AWESOME, @minrk \o/

**Pull request merged**